src/common/leaky_bucket.cpp

```diff
@@ -79,29 +86,58 @@ double LeakyBucket::Bucket::Add(double tokens) {
     }
 }
 
-void LeakyBucket::Bucket::Leak(uint64_t intervalUs) {
+void LeakyBucket::Bucket::Leak() {
     if (avg == 0) {
         level = 0;
         burstLevel = 0;
         return;
     }
 
-    double leak = static_cast<double>(avg) * intervalUs /
-                  TimeUtility::MicroSecondsPerSecond;
-    level = std::max(level - leak, 0.0);
+    if (level == 0) {
+       bigBucketEmptyTimes++;
+       if (bigBucketEmptyTimes >= FLAGS_bucketEmptyTimes) {
+            levelInitial = true;
+            level = avg - static_cast<double>(avg) * FLAGS_bucketLeakIntervalMs
+              / TimeUtility::MilliSecondsPerSecond;
+            bigBucketEmptyTimes = 0;
```

Comment:
why not just return here?

 if (levelInitial)
        return;

Reply:
when there is no io, we do not need to leak.